economics/cpi.py

```diff
@@ -18,89 +18,68 @@
 
 import datetime
 import collections
-import economics.data as data
-from economics.datastructures import MapDict
+import requests
 
 CPIResult = collections.namedtuple('CPI', 'date value')
 
-class CPI(object):
+
+class CPI:
     """
     Provides a Pythonic interface to Consumer Price Index data packages
     """
 
-    def __init__(self, datapackage='http://data.okfn.org/data/cpi/',
-                 country=None):
+    def __init__(self, country="all"):
         """
-        Initialise a CPI instance. Default data package location is the cpi
-        data on http://data.okfn.org
+        Initialise a CPI instance.
         """
 
-        # Store datapackage and country as instance variables
-        self.datapackage = datapackage
-        self.country = country
-
         # Initialise empty data structures
-        self.data = MapDict()
+        self.data = collections.ChainMap()
 
         # Load the data into the data structures
-        self.load()
+        self.load(country)
+        self.country = country
 
-    def load(self):
+    def load(self, country="all"):
         """
-        Load data with the data from the datapackage
+        Load data
         """
+        u = ("https://api.worldbank.org/v2/countries/{}/indicators/CPTOTSAXN"
+             "?format=json&per_page=10000").format(country)
```

Comment:
The original idea was to use [data packages](https://frictionlessdata.io/specs/data-package/) to build up the economics package (and that was only the CPI package then.

Doing it this way is also fine. It's not as reusable in other software (which data packages are) and also creates a small burden on maintenance if we start pulling in other sources but as long as we're fulfilling requirements (which we aren't now) then we should go for the approach that works best and is most up to date. :+1:

Reply:
as per #2 , the underlying data appears to have changed. The new data package also does not have current data. I did consider allowing sources to be different world bank metrics, but that might be one for the future.